pkgs/by-name/nh/nhost-cli/package.nix

```diff
@@ -0,0 +1,37 @@
+{
+  lib,
+  buildGoModule,
+  fetchFromGitHub,
+}:
+
+buildGoModule rec {
+  pname = "nhost-cli";
+  version = "1.18.1";
+
+  src = fetchFromGitHub {
+    owner = "nhost";
+    repo = "cli";
+    rev = "v${version}";
+    hash = "sha256-B0kkbRwL808+p6WwvitxY+6FvnjkNN7NFSSYpNJNOrk=";
+  };
+
+  vendorHash = null;
+
+  ldflags = [
+    "-s"
+    "-w"
+    "-X=main.Version=v${version}"
+  ];
+
+  postInstall = ''
+    mv $out/bin/cli $out/bin/${meta.mainProgram}
```

Comment:
mainProgram shouldn't be used like this

Reply:
Could you please provide some relevant documents? In fact, this usage is recommended by @drupol, considering this `postInstall` line is to make the binary file conform to `meta.mainProgram` and `pname`, I decided to accept it.

Reply:
Normally changing meta shouldn't necessarily trigger a rebuild. Also there is a tracking issue of removing overly used substitution of pname because when migrating to finalAttrs, they often break or are no longer correct, as almost any change to pname (eg adding `-headless`) would break the package or cause unintended side effects. I would say this usage of meta.mainProgram falls under the same.

Also right now an override to the package would not update this postInstall, which is probably also not really intended. 

And I myself wouldn't recommend this pattern.

Reply:
Thank you for your reply! As I understand it, the key to minimize the rebuilds caused by changes to non-critical parts as much as possible. I will open another PR to fix this issue.

Reply:
Thanks. I won't recommend this pattern from now on.